modules/directives/src/main/scala/scala/build/preprocessing/directives/ComputeVersion.scala

```diff
@@ -0,0 +1,45 @@
+package scala.build.preprocessing.directives
+
+import scala.build.EitherCps.{either, value}
+import scala.build.Ops.EitherOptOps
+import scala.build.Positioned
+import scala.build.directives.{
+  DirectiveDescription,
+  DirectiveExamples,
+  DirectiveGroupName,
+  DirectiveLevel,
+  DirectiveUsage,
+  HasBuildOptions
+}
+import scala.build.errors.BuildException
+import scala.build.options.{BuildOptions, ComputeVersion => cv, SourceGeneratorOptions}
+import scala.cli.commands.SpecificationLevel
+
+@DirectiveGroupName("Compute Version")
+@DirectiveExamples("//> using computeVersion git:tag")
+@DirectiveExamples("//> using computeVersion command:'echo 1.0.0'")
```

Comment:
Perhaps these examples could be improved, for example - reading a version from a file.

```
//> using computeVersion command:'cat project.version'"
```

If you running `echo 1.0.0`, it might not make much sense. It would be better to use the `version` option.